plugins/fluent-bit/src/plugin.cpp

```diff
@@ -0,0 +1,447 @@
+      return false;
+    }
+    // Apply user-provided plugin properties.
+    for (const auto& [key, value] : properties) {
+      TENZIR_DEBUG("setting {} plugin option: {}={}", plugin, key, value);
+      if (flb_input_set(ctx_, ffd, key.c_str(), value.c_str(), nullptr) != 0) {
+        TENZIR_ERROR("failed to set {} plugin option: {}={}", plugin, key,
+                     value);
+        return false;
+      }
+    }
+    return true;
+  }
+
+  auto output(const std::string& plugin, const property_map& properties = {},
+              struct flb_lib_out_cb* callback = nullptr) -> bool {
+    auto ffd = flb_output(ctx_, plugin.c_str(), callback);
+    if (ffd < 0) {
+      TENZIR_ERROR("failed to setup {} output plugin ({})", plugin, ffd);
+      return false;
+    }
+    // Apply user-provided plugin properties.
+    for (const auto& [key, value] : properties) {
+      TENZIR_DEBUG("setting {} plugin option: {}={}", plugin, key, value);
+      if (flb_output_set(ctx_, ffd, key.c_str(), value.c_str(), nullptr) != 0) {
+        TENZIR_ERROR("failed to set {} plugin option: {}={}", plugin, key,
+                     value);
+        return false;
+      }
+    }
+    return true;
+  }
+
+  auto start() -> bool {
+    auto ret = flb_start(ctx_);
+    if (ret == 0)
+      return true;
+    TENZIR_ERROR("failed to start engine ({})", ret);
+    return false;
+  }
+
+  flb_ctx_t* ctx_{nullptr};
+  std::chrono::milliseconds poll_interval_{};
+  shared_state state_{.buf = nullptr, .len = 0};
+  std::string buffer_{};
+};
+
+class fluent_bit_operator final : public crtp_operator<fluent_bit_operator> {
+public:
+  fluent_bit_operator() = default;
+
+  fluent_bit_operator(operator_args args, record config)
+    : args_{std::move(args)}, config_{std::move(config)} {
+  }
+
+  auto operator()(operator_control_plane& ctrl) const
+    -> generator<table_slice> {
+    auto engine = engine::make_source(args_, config_);
+    if (not engine) {
+      diagnostic::error("failed to create Fluent Bit engine")
+        .emit(ctrl.diagnostics());
+      co_return;
+    }
+    while (engine->running()) {
+      auto result = table_slice{};
+      auto parse_fluentbit = [&result](std::string_view data) {
+        // TODO: create table slices from JSON. The alternative would be
+        // switching the format to MsgPack, but this is perhaps v2 when things
+        // are too slow.
+        //
+        // Speaking with Dominik, we could use simdjson directly here to get the
+        // second array element of the Fluent Bit message [timestamp, {..}],
+        // using something like on_demand::dom::parse. Once we're ready to
+        // extract the contained object, we should go to Jannis as he's
+        // currently reworking the JSON parser. What we need is a library
+        // utility that bakes us a table slice given a JSON object and builder.
+        result = {};
+        return true;
+      };
```

Comment:
@jachris maybe this is helpful for you to know while reworking the JSON parser.

Reply:
It sounds like this should be easy to implement with a bit of refactoring.

Reply:
May I ask you to ping me once you have completed the adaptive builder changes?